File: DevExpress-Demo/CS/GanttDemo.Wpf/Modules/Examples/PartsProductionViewModel.cs
using System;
using System.Collections.Generic;

namespace GanttDemo.Examples {
    public class PartsProductionViewModel {
        public PartsProductionViewModel() {
            #region initialization
            var startDate = DateTime.Now.Date.AddDays(3);
            Parts = new List<PartItem> {
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(10),
                    Name = "Part 1",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(1) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(11),
                    Name = "Part 2",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(2) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(4),
                    Name = "Part 3",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(3) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(3),
                    Name = "Part 4",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(4) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(21),
                    Name = "Part 5",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(5) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(12),
                    Name = "Part 6",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(6) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(10),
                    Name = "Part 7",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(7) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(12),
                    Name = "Part 8",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(8) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(20),
                    Name = "Part 9",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(9) }
                },
                new PartItem() {
                    StartProductionDate = startDate,
                    FinishProductionDate = startDate.AddDays(4),
                    Name = "Part 10",
                    MachineTools = new List<MachineToolComponentLink>() { new MachineToolComponentLink(10) }
                },
            };

            MachineTools = new List<MachineToolItem> {
                new MachineToolItem { Name = "Machine tool 1", Id = 1, WorkshopName = "Workshop 1" },
                new MachineToolItem { Name = "Machine tool 2", Id = 2, WorkshopName = "Workshop 1" },
                new MachineToolItem { Name = "Machine tool 3", Id = 3, WorkshopName = "Workshop 1" },
                new MachineToolItem { Name = "Machine tool 1", Id = 4, WorkshopName = "Workshop 2" },
                new MachineToolItem { Name = "Machine tool 2", Id = 5, WorkshopName = "Workshop 2" },
                new MachineToolItem { Name = "Machine tool 3", Id = 6, WorkshopName = "Workshop 2" },
                new MachineToolItem { Name = "Machine tool 4", Id = 7, WorkshopName = "Workshop 2" },
                new MachineToolItem { Name = "Machine tool 1", Id = 8, WorkshopName = "Workshop 3" },
                new MachineToolItem { Name = "Machine tool 2", Id = 9, WorkshopName = "Workshop 3" },
                new MachineToolItem { Name = "Machine tool 3", Id = 10, WorkshopName = "Workshop 3" }
            };
            #endregion
        }

        public IEnumerable<PartItem> Parts { get; private set; }
        public IEnumerable<MachineToolItem> MachineTools { get; private set; }
    }

    public class MachineToolItem {
        public string Name { get; set; }
        public string WorkshopName { get; set; }
        public int Id { get; set; }
        public string DisplayName { get { return WorkshopName + " / " + Name; } }
    }

    public class MachineToolComponentLink {
        public MachineToolComponentLink() : this(0) { }
        public MachineToolComponentLink(int id) {
            this.MachineToolId = id;
        }
        public int MachineToolId { get; set; }
    }

    public class PartItem {
        public string Name { get; set; }
        public DateTime StartProductionDate { get; set; }
        public DateTime FinishProductionDate { get; set; }
        public List<MachineToolComponentLink> MachineTools { get; set; }
    }
}

File: DevExpress-Demo/VB/GanttDemo.Wpf/Modules/Examples/PartsProductionViewModel.vb
Imports System.Collections.Generic

Namespace GanttDemo.Examples

    Public Class PartsProductionViewModel

        Private _Parts As IEnumerable(Of GanttDemo.Examples.PartItem), _MachineTools As IEnumerable(Of GanttDemo.Examples.MachineToolItem)

        Public Sub New()
#Region "initialization"
            Dim startDate = Date.Now.Date.AddDays(3)
            Parts = New List(Of PartItem) From {New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(10), .Name = "Part 1", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(1)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(11), .Name = "Part 2", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(2)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(4), .Name = "Part 3", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(3)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(3), .Name = "Part 4", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(4)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(21), .Name = "Part 5", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(5)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(12), .Name = "Part 6", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(6)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(10), .Name = "Part 7", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(7)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(12), .Name = "Part 8", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(8)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(20), .Name = "Part 9", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(9)}}, New PartItem() With {.StartProductionDate = startDate, .FinishProductionDate = startDate.AddDays(4), .Name = "Part 10", .MachineTools = New List(Of MachineToolComponentLink)() From {New MachineToolComponentLink(10)}}}
            MachineTools = New List(Of MachineToolItem) From {New MachineToolItem With {.Name = "Machine tool 1", .Id = 1, .WorkshopName = "Workshop 1"}, New MachineToolItem With {.Name = "Machine tool 2", .Id = 2, .WorkshopName = "Workshop 1"}, New MachineToolItem With {.Name = "Machine tool 3", .Id = 3, .WorkshopName = "Workshop 1"}, New MachineToolItem With {.Name = "Machine tool 1", .Id = 4, .WorkshopName = "Workshop 2"}, New MachineToolItem With {.Name = "Machine tool 2", .Id = 5, .WorkshopName = "Workshop 2"}, New MachineToolItem With {.Name = "Machine tool 3", .Id = 6, .WorkshopName = "Workshop 2"}, New MachineToolItem With {.Name = "Machine tool 4", .Id = 7, .WorkshopName = "Workshop 2"}, New MachineToolItem With {.Name = "Machine tool 1", .Id = 8, .WorkshopName = "Workshop 3"}, New MachineToolItem With {.Name = "Machine tool 2", .Id = 9, .WorkshopName = "Workshop 3"}, New MachineToolItem With {.Name = "Machine tool 3", .Id = 10, .WorkshopName = "Workshop 3"}}
#End Region
        End Sub

        Public Property Parts As IEnumerable(Of PartItem)
            Get
                Return _Parts
            End Get

            Private Set(ByVal value As IEnumerable(Of PartItem))
                _Parts = value
            End Set
        End Property

        Public Property MachineTools As IEnumerable(Of MachineToolItem)
            Get
                Return _MachineTools
            End Get

            Private Set(ByVal value As IEnumerable(Of MachineToolItem))
                _MachineTools = value
            End Set
        End Property
    End Class

    Public Class MachineToolItem

        Public Property Name As String

        Public Property WorkshopName As String

        Public Property Id As Integer

        Public ReadOnly Property DisplayName As String
            Get
                Return WorkshopName & " / " & Name
            End Get
        End Property
    End Class

    Public Class MachineToolComponentLink

        Public Sub New()
            Me.New(0)
        End Sub

        Public Sub New(ByVal id As Integer)
            MachineToolId = id
        End Sub

        Public Property MachineToolId As Integer
    End Class

    Public Class PartItem

        Public Property Name As String

        Public Property StartProductionDate As Date

        Public Property FinishProductionDate As Date

        Public Property MachineTools As List(Of MachineToolComponentLink)
    End Class
End Namespace
